src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseReport.java

```diff
@@ -0,0 +1,607 @@
+    if (row != null) return row;
+    rowNum -= (toDateTwentyFour.size() + 4); // list + (blank, 2 headers, total)
+
+    row = maybeInList("Tissue", labelListSlides, toDateSlides, rowNum);
+    if (row != null) return row;
+    rowNum -= (toDateSlides.size() + 4); // list + (blank, 2 headers, total)
+
+    if (rowNum == 0) {
+      return makeBlankRow();
+    } else if (rowNum == 1) {
+      return makeAlmostBlankRow(String.format("Samples Received This Month (%s - %s)", start, end));
+    }
+    rowNum -= 2;
+
+    row = maybeInList(TimePoint.RANDOMIZATION.getKey(), labelListLong, thisMonthRandom, rowNum);
+    if (row != null) return row;
+    rowNum -= (thisMonthRandom.size() + 4); // list + (blank, 2 headers, total)
+
+    row = maybeInList(TimePoint.SIX.getKey(), labelListShort, thisMonthSix, rowNum);
+    if (row != null) return row;
+    rowNum -= (thisMonthSix.size() + 4); // list + (blank, 2 headers, total)
+
+    row = maybeInList(TimePoint.TWELVE.getKey(), labelListLong, thisMonthTwelve, rowNum);
+    if (row != null) return row;
+    rowNum -= (thisMonthTwelve.size() + 4); // list + (blank, 2 headers, total)
+
+    row = maybeInList(TimePoint.EIGHTEEN.getKey(), labelListShort, thisMonthEighteen, rowNum);
+    if (row != null) return row;
+    rowNum -= (thisMonthEighteen.size() + 4); // list + (blank, 2 headers, total)
+
+    row = maybeInList(TimePoint.TWENTY_FOUR.getKey(), labelListLong, thisMonthTwentyFour, rowNum);
+    if (row != null) return row;
+    rowNum -= (thisMonthTwentyFour.size() + 4); // list + (blank, 2 headers, total)
+
+    row = maybeInList("Tissue", labelListSlides, thisMonthSlides, rowNum);
+    if (row != null) return row;
+    rowNum -= (thisMonthSlides.size() + 4); // list + (blank, 2 headers, total)
+
+    if (rowNum == 0) {
+      return makeBlankRow();
+    } else if (rowNum == 1) {
+      return makeAlmostBlankRow(String.format("Total Inventory Available (%s)", new SimpleDateFormat(DATE_FORMAT).format(new Date())));
+    }
+    rowNum -= 2;
+
+    row = maybeInList("Total in Inventory", labelListInventory, inventoryAvailable, rowNum);
+    if (row != null) return row;
+    rowNum -= (inventoryAvailable.size() + 4); // list + (blank, 2 headers, total)
+
+    if (rowNum == 0) {
+      return makeBlankRow();
+    } else if (rowNum == 1) {
+      String[] blank = makeBlankRow();
+      blank[0] = "Samples with mismatched alias and cryovial/slide number";
+      return blank;
+    }
+
+    row = maybeSampleAlias(barcodeAndNameCodingMismatch, rowNum);
+    if (row != null) return row;
+    rowNum -= barcodeAndNameCodingMismatch.size();
+
+    if (rowNum == 0) {
+      return makeBlankRow();
+    } else if (rowNum == 1) {
+      String[] blank = makeBlankRow();
+      blank[0] = "Slides with non-slide timepoint in alias";
+      return blank;
+    }
+
+    row = maybeSampleAlias(slidesWithoutSlideTimepoint, rowNum);
+    if (row != null) return row;
+    rowNum -= slidesWithoutSlideTimepoint.size();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "rowNum". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)